SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -176,7 +176,27 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject&
 		// Scoped to exclude OnRep callbacks which are already tracked per OnRep function
 		SCOPE_CYCLE_COUNTER(STAT_ReaderApplyPropertyUpdates);
 
-		for (uint32 FieldId : UpdatedIds)
+		// If we are applying initial data, they must have come from a ComponentData (as it currently stands).
```

Comment:
Is bIsInitialData the same as it being a component data?
We want to run this whenever it came from a component data not just when it's the first checkout.

Reply:
Yeah, as it is currently bIsInitialData iff we are applying ComponentData.
But I recently read an article about not using boolean flags, and became convinced they are evil, so we should probably make things a bit nicer in the future.